### ISSUE TITLE
Training never finish

### ISSUE THREAD

**Issue by nukaemon** [JaesikKim/HiG2Vec#5]
Hello @JaesikKim

When I executed run_embedding.sh on the input files I prepared, it kept running without any error but never finished.
The process was left without termination but it seems like in sleep status after all epoches.
WhenI executed run_embedding.sh on the example files in data direcotry, it finished properly.

The differences I saw betwen when using the example input files and when using my own input files are:

- When using the example input file, a pth file of each epoch was removed every time befor moving to the next epoch. Also I saw GOonly_200dim.pth was progressively updated at each epoch.
- When using my own input files, pth files are not removed and accumulated as GOonly_200dim.pth.0, GOonly_200dim.pth.1 and so on. I also confirmed GOonly_200dim.pth was never created during the process.

- When using the example input files, I saw lines starting with "json_stats:" between progress bars of epochs. This is also true after burn-in step.
```
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 1/1 [00:00<00:00,  3.16it/s]
json_stats: {"epoch": 0, "elapsed": 0.0054249875247478485, "loss": 3.931826683098391, "sqnorm_min": 0.0006856000455276103, "sqnorm_avg": 0.0008176872341644401, "sqnorm_max": 0.0008731740667575559}
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 1/1 [00:00<00:00, 86.36it/s]
json_stats: {"epoch": 1, "elapsed": 0.0029672086238861084, "loss": 3.931681898676436, "sqnorm_min": 0.0005846760973066063, "sqnorm_avg": 0.0008216559489371129, "sqnorm_max": 0.0008816333755467856}
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 1/1 [00:00<00:00, 74.19it/s]
json_stats: {"epoch": 2, "elapsed": 0.0018315277993679047, "loss": 3.931514766418724, "sqnorm_min": 0.0005188694821587003, "sqnorm_avg": 0.0008289513472264196, "sqnorm_max": 0.0009098924743264128}
```

- When using my own input files, I didn't see any lines starting with "json_stats:". This is also true after burn-in step.
```
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 158/158 [00:11<00:00, 14.29it/s]
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 158/158 [00:14<00:00, 10.77it/s]
Burn in negs=50, lr=0.003
100%|█████████████████████████████████| 158/158 [00:24<00:00,  6.42it/s]
Burn in negs=50, lr=0.003
```

Do you come up with any reason why I get this problem?

**Comment by JaesikKim** [comment]
hmm.. not really.. Can you share your input file to me (file after running transitive closure)? I can test on my env. and try to fix bugs, if possible. You can share here or send to my email (jaesik.kim@pennmedicine.upenn.edu)
And would be better to use -eval_each 10 because this value means how often checkpoints should be stored.

**Comment by nukaemon** [comment]
Sorry for my late reply.
I have just sent my input file to your email.
It may be a problem of my way of generating the input files, but I don't know what is wrong.

I tried running with '-eval_each 10' as you suggested, and now I can see the lines starting with 'jaon_stats' which I didn't see previously.
However, training process never finished again, leaving pth files of every 10 epoches.
I paste below a part of standard error messages during and after burn-in process:

```
.
.
.
json_stats: {"epoch": 15, "elapsed": 1.7955227121710777, "loss": 3.9263721691507962}
Burn in negs=50, lr=0.003
json_stats: {"epoch": 16, "elapsed": 1.757146067917347, "loss": 3.925995571922804}
Burn in negs=50, lr=0.003
json_stats: {"epoch": 17, "elapsed": 1.8817334808409214, "loss": 3.9256210089473806}
Burn in negs=50, lr=0.003
json_stats: {"epoch": 18, "elapsed": 1.800656445324421, "loss": 3.925235025567851}
Burn in negs=50, lr=0.003
json_stats: {"epoch": 20, "elapsed": 1.9878743886947632, "loss": 3.9230784827439513}
json_stats: {"epoch": 21, "elapsed": 1.950884137302637, "loss": 3.898324795798827}
json_stats: {"epoch": 22, "elapsed": 2.0491686314344406, "loss": 3.8743761797855707}
json_stats: {"epoch": 23, "elapsed": 1.995950624346733, "loss": 3.850105552062385}
json_stats: {"epoch": 24, "elapsed": 1.9556585997343063, "loss": 3.825481953640994}
json_stats: {"epoch": 25, "elapsed": 1.913625556975603, "loss": 3.8009482733074416}
.
.
.
```

**Comment by genggengcss** [comment]
Hi, I have the same problem. Do any new solutions are provided?

**Comment by JaesikKim** [comment]
@genggengcss Hi,
Last time in the conversation with @nukaemon by email, I suggested one solution for this. 
Please see the message I sent below.
For better organization of this issue, I will close this one.
If you have still this issue, please raise a new issue.



With your input files, my env was working well, and it was terminated well.

1) First attached image is the end of log for GO embedding.

2) Next attached image is the end of log for Gene embedding.

3) My running setting is below.

python3 embedGO.py \
       -dim 200 \
       -lr 0.3 \
       -epochs 100 \
       -negs 50 \
       -batchsize 50 \
       -burnin 20 \
       -eval_each 10 \
       -gpu 0 \
       -manifold poincare \
       -dset data/my_input/GO_closure_4.tsv \
       -checkpoint result/GOonly_200dim.pth \
       -fresh \
       -sparse

python3 embedGene.py \
       -dim 200 \
       -lr 1 \
       -epochs 100 \
       -negs 50 \
       -batchsize 50 \
       -burnin 20 \
       -gpu 0 \
       -manifold poincare \
       -dset data/my_input/GOA.tsv \
       -pretrain result/GOonly_200dim.pth \
       -checkpoint result/hig2vec_200dim.pth \
       -sym \
       -fresh \
       -sparse \
       -finetune

python3 combine.py \
       -GOonly result/GOonly_200dim.pth \
       -gene_embedding result/hig2vec_200dim.pth


And in case, I’m attaching the final output file.

If your running code still doesn’t work, can you try with new env. and install my code again?

Hope you resolve this issue..
![image](https://user-images.githubusercontent.com/46479633/225128958-510597ed-4dba-42f6-b00f-3b25beb6d9a9.png)
![image](https://user-images.githubusercontent.com/46479633/225129003-bc0dd83b-0857-409e-8b4b-64928ae2ea93.png)
